godot-core/src/builtin/dictionary.rs

```diff
@@ -441,6 +441,23 @@ impl<'a> DictionaryIter<'a> {
         Some((key, value))
     }
 
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let total_length = self.dictionary.len();
+        if total_length == 0 {
+            return (0, Some(0));
+        }
+        if self.is_first {
+            return (total_length, Some(total_length));
+        }
+        let prev_idx = self
+            .dictionary
+            .keys_array()
+            .find(&self.last_key.clone().unwrap(), None)
+            .unwrap();
```

Comment:
I'm worried about perf here -- `keys_array()` allocates a new array and copies all the keys, which is O(n).

`find()` is also O(n).


This defeats the idea of `size_hint` giving a fast, O(1) hint that can then be used for optimization.

Reply:
Oh, sorry! Quite new to the language's quirks.

I thought about it too, and my first thought was using a persistent index like an `Array`'s, but I thought `Dictionary` is inherently "unordered". I suppose it just can't guarantee order, but I'll implement an index so it all stays O(1) :)

Reply:
Copied implementation from `Array`, to remain in O(1).